Emby.Server.Implementations/Library/MediaStreamSelector.cs

```diff
@@ -60,13 +62,19 @@ public static class MediaStreamSelector
                 // If no subtitles of preferred language available, use default behaviour.
                 if (!preferredLanguages.Contains(audioTrackLanguage, StringComparison.OrdinalIgnoreCase))
                 {
-                    stream = sortedStreams.FirstOrDefault(x => preferredLanguages.Contains(x.Language, StringComparison.OrdinalIgnoreCase)) ??
-                        sortedStreams.FirstOrDefault(x => x.IsExternal || x.IsForced || x.IsDefault);
+                    // In this case, we don't want to prefer forced subtitles, but full subtitles, so sort again
+                    stream = sortedStreams
+                        .OrderByDescending(x => preferredLanguages.Contains(x.Language, StringComparison.OrdinalIgnoreCase))
+                        .ThenByDescending(x => x.IsExternal)
+                        .ThenByDescending(x => x.IsDefault)
+                        .ThenByDescending(x => !x.IsForced)
+                        .ThenByDescending(x => preferredCodecs.Contains(x.Codec, StringComparison.OrdinalIgnoreCase))
+                        .FirstOrDefault();
                 }
                 else
                 {
-                    // Respect forced flag.
-                    stream = sortedStreams.FirstOrDefault(x => x.IsForced);
+                    // Respect forced flag, but make sure to match the language
+                    stream = sortedStreams.FirstOrDefault(x => x.IsForced && string.Equals(x.Language, audioTrackLanguage, StringComparison.OrdinalIgnoreCase));
```

Comment:
Just to make sure, this means Forced will only be considered when the audio track language is the same, correct?

Reply:
Yes, I would say generally you would only want forced subtitles in the same language as the audio, if that language is your preferred subtitle language.

Reply:
I agree. To my knowledge that is also in line with the Matroska spec.